Rs/intro (#121)
neuralmagic/docs: src/content/get-started/use-a-model/nlp-text-classification.mdx

---
title: "NLP Text Classification"
metaTitle: "Use a Text Classification Model"
metaDescription: "Use a NLP Text Classification Model with DeepSparse for faster and cheaper inference on CPUs"
index: 1000
---

# Use a Text Classification Model

This page explains how to run a trained model with DeepSparse for NLP inside a Python API called `Pipelines.`

> `Pipelines` wraps key utilities around the DeepSparse for easy testing and deployment.

Reviewer: ```suggestion
`Pipelines` wraps key utilities around DeepSparse for easy testing and deployment.
```

> The text classification `Pipeline`, for example, wraps an NLP model with the proper preprocessing and postprocessing pipelines, such as tokenization.

Reviewer: ```suggestion
The text classification `Pipeline`, for example, wraps an NLP model with the proper pre-processing and post-processing pipelines, such as tokenization.
```

With the text classification model set up, the model can be passed into a DeepSparse `Pipeline` utilizing the `model_path` argument.
The SparseZoo stub for the sparse-quantized DistilBERT model given at the beginning is used in the sample code below.

> The `Pipeline` automatically downloads the necessary files for the model from the SparseZoo and compiles them on your local machine in the DeepSparse engine.

Reviewer: ```suggestion
The `Pipeline` automatically downloads the necessary files for the model from the SparseZoo and compiles them on your local machine in DeepSparse.
```
